app/scripts/directives/mobileClientBuildsList.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileClientBuildsList', {
+    controller: [
+      '$timeout',
+      '$anchorScroll',
+      '$location',
+      'APIService',
+      'DataService',
+      MobileClientBuildsList
+    ],
+    bindings: {
+      mobileClient: '<'
+    },
+    templateUrl: 'views/mobile-client-builds-list.html'
+  });
+
+  function MobileClientBuildsList(
+    $timeout,
+    $anchorScroll,
+    $location,
+    APIService,
+    DataService
+  ) {
+    var ctrl = this;
+    var buildConfigsVersion = APIService.getPreferredVersion('buildconfigs');
+    var watches = [];
+    var anchor;
+
+    ctrl.$onChanges = function(changes) {
```

Comment:
You have a lot of similarities between the `$onChanges` functions, aside from different variable names.  Possible to factor any of it out into shared utility functions via a service?

Reply:
I left this as is as it just attaching event handlers. I am not sure there is too much benefit from having this within a service.